student.rb

```diff
@@ -1,12 +1,20 @@
 require './person'
+require './classroom'
 
 class Student < Person
-  def initialize(classroom, id, age, name = 'Unknown', parent_permission: true)
-    super(id, name, age, parent_permission)
+  attr_accessor :name, :classroom
+
+  def initialize(classroom, age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
 
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def add_classroom=(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
```

Comment:
- In here, you should explicitly add the classroom itself to the student, instead of having to call the add_student method to improve the readability of your code. kindly update this to `classroom.students.push(self)`